XLS-0056d-batch/README.md

```diff
@@ -0,0 +1,503 @@
+
+This field must be provided if more than one account has inner transactions included in the `Batch`. In that case, this field must contain signatures from all accounts whose inner transactions are included, excluding the account signing the outer transaction (if applicable).
+
+If all inner transactions are from the same account, this field must be omitted, as the `TxnSignature` field (the normal transaction signature field) will be used instead.
+
+Each object in this array contains the following fields:
+
+|FieldName | Required? | JSON Type | Internal Type |
+|:---------|:-----------|:---------------|:------------|
+|`Account`|✔️|`string`|`STAccount`|
+|`SigningPubKey`| |`string`|`STBlob`|
+|`TxnSignature`| |`string`|`STBlob`|
+|`Signers`| |`array`|`STArray`|
+
+Either the `SigningPubKey` and `TxnSignature` fields must be included, or the `Signers` field.
+
+#### 2.5.1. `Account`
+
+This is an account that has at least one inner transaction.
+
+#### 2.5.2. `SigningPubKey` and `TxnSignature`
+
+These fields are included if the account is signing with a single signature (as opposed to multi-sign). They sign the `Flags` and `TxnIDs` fields.
+
+#### 2.5.3. `Signers`
+
+This field is included if the account is signing with multi-sign (as opposed to a single signature). It operates equivalently to the [`Signers` field](https://xrpl.org/docs/references/protocol/transactions/common-fields/#signers-field) used in standard transaction multi-sign. This field holds the signatures for the `Flags` and `TxnIDs` fields.
+
+### 2.6. Metadata
+
+The inner transactions will be committed separately to the ledger and will therefore have separate metadata. This is to ensure better backwards compatibility for legacy systems, so they can support `Batch` transactions without needing any changes to their systems.
+
+For example, a ledger that only has one `Batch` transaction containing 2 inner transactions would look like this:
+```
+[
+  OuterTransaction,
+  InnerTransaction1,
+  InnerTransaction2
+]
+```
+
+#### 2.6.1. Outer Transactions
+
+Each outer transaction will only contain the metadata for its sequence and fee processing, not for the inner transaction processing.
+
+There will also be a list of which transactions were actually processed, which is useful for the `ONLYONE` and `UNTILFAILURE` modes, since those may only process a subset of transactions, and for debugging with all modes. This section will be called `BatchExecutions`.
+
+It will contain a list of objects that have the following fields for every transaction that is processed (successfully or unsuccessfully):
+
+|FieldName | Required? | JSON Type | Internal Type |
+|:---------|:-----------|:---------------|:------------|
+|`TransactionHash`|✔️|`string`|`STUInt256`|
+|`InnerResult`|✔️|`string`|`STBlob`|
```

Comment:
doesn't it also have `sfTransactionType`?

Reply:
It currently has that in place of `TransactionHash`. I'd prefer `TransactionHash`, personally.